juju/client/connector.py

```diff
@@ -67,6 +67,13 @@ async def connect(self, **kwargs):
             for macaroon in kwargs.pop('macaroons'):
                 jar.set_cookie(go_to_py_cookie(macaroon))
         self._connection = await Connection.connect(**kwargs)
+        controller = self.jujudata.controllers()[
+            self.jujudata.current_controller()
+        ]
+        self._connection.endpoints = [
+            (e, controller["ca-cert"])
+            for e in controller["api-endpoints"]
+        ]
```

Comment:
@SimonRichardson I would like to set the endpoints within the Connection, but the problem is that if I get the controller from the connection, it enters into a loop.

If you come up with a better solution, let me know :-)